### PR TITLE
Build improvements

### PR DESCRIPTION
**Pull request by hohle**
Creates the intermediate `build` directory before changing into it.

Conditionally includes `malloc.h` on Windows (`malloc(3)` is  part of `stdlib.h` in C99).

Tested on:

* OS X 10.11.5
* FreeBSD 10.3
* CentOS 7.2

### CODE REVIEW

File: tools/ionizer/options.c [hunk 1]
```diff
@@ -10,8 +10,8 @@
 #include <stdio.h>
 #ifdef WIN
 #include <io.h>
-#endif
 #include <malloc.h>
+#endif
```
Comment (almann) [inline tools/ionizer/options.c]: My understanding is that `malloc.h` is an archaic header for Linux systems, and is probably not on MSVC.  Perhaps we should eliminate this include all together or prefer `stdlib.h`

File: tools/ionsymbols/options.c [hunk 2]
```diff
@@ -10,8 +10,8 @@
 #include <stdio.h>
 #ifdef WIN
 #include <io.h>
-#endif
 #include <malloc.h>
+#endif
```
Comment (almann) [inline tools/ionsymbols/options.c]: Same as above.

**Comment by almann** [comment]
This looks good, thanks for doing this--I don't think `malloc.h` exists in MSVC, I think we can kill it.

File: tools/ionizer/options.c (hunk 1 above)
Comment (hohle) [inline tools/ionizer/options.c]: I left it in based on the [MSDN `malloc` docs](https://msdn.microsoft.com/en-us/library/6ewkz86d.aspx), but I have no way to test it.

Comment (almann) [inline tools/ionizer/options.c]: Fair enough.

**Pull request merged**